src/net_processing.cpp

```diff
@@ -479,6 +473,13 @@ struct Peer {
     /** Same id as the CNode object for this peer */
     const NodeId m_id{0};
 
+    /** Protects misbehavior data members */
+    Mutex m_misbehavior_mutex;
```

Comment:
I think it's easier to reason about having both misbehavior fields (`m_misbehavior_score` and `m_should_discourage`) under the same mutex since they're so closely related.